sdks/java/io/tika/src/main/java/org/apache/beam/sdk/io/tika/TikaIO.java

```diff
@@ -275,36 +206,49 @@ public void populateDisplayData(DisplayData.Builder builder) {
             .add(DisplayData.item("inputMetadata", sb.toString())
             .withLabel("Input Metadata"));
       }
-      if (Boolean.TRUE.equals(getParseSynchronously())) {
-        builder
-          .add(DisplayData.item("parseMode", "synchronous")
-            .withLabel("Parse Mode"));
-      } else {
-        builder
-          .add(DisplayData.item("parseMode", "asynchronous")
-            .withLabel("Parse Mode"));
-        builder
-          .add(DisplayData.item("queuePollTime", getQueuePollTime().toString())
-            .withLabel("Queue Poll Time"))
-        .add(DisplayData.item("queueMaxPollTime", getQueueMaxPollTime().toString())
-          .withLabel("Queue Max Poll Time"));
-      }
-      Integer minTextLen = getMinimumTextLength();
-      if (minTextLen != null && minTextLen > 0) {
-        builder
-        .add(DisplayData.item("minTextLen", getMinimumTextLength().toString())
-          .withLabel("Minimum Text Length"));
-      }
-      if (Boolean.TRUE.equals(getReadOutputMetadata())) {
-        builder
-          .add(DisplayData.item("readOutputMetadata", "true")
-            .withLabel("Read Output Metadata"));
-      }
     }
 
-    @Override
-    protected Coder<String> getDefaultOutputCoder() {
-      return StringUtf8Coder.of();
+    private static class ParseToStringFn extends DoFn<ReadableFile, ParseResult> {
+
+      private static final long serialVersionUID = 6837207505313720989L;
+      private TikaIO.ParseAll spec;
+      ParseToStringFn(TikaIO.ParseAll spec) {
+        this.spec = spec;
+      }
+      @ProcessElement
+      public void processElement(ProcessContext c) throws IOException {
+        ReadableFile file = c.element();
+        InputStream stream = Channels.newInputStream(file.open());
+
+        final InputStream is = TikaInputStream.get(stream);
+        TikaConfig tikaConfig = null;
+        if (spec.getTikaConfigPath() != null) {
+          try {
+            tikaConfig = new TikaConfig(spec.getTikaConfigPath().get());
+          } catch (TikaException | SAXException e) {
+            throw new IOException(e);
+          }
+        }
+        final Parser parser = tikaConfig == null ? new AutoDetectParser()
+            : new AutoDetectParser(tikaConfig);
+        final ParseContext context = new ParseContext();
+        context.set(Parser.class, parser);
+        org.apache.tika.metadata.Metadata tikaMetadata = spec.getInputMetadata() != null
+          ? spec.getInputMetadata() : new org.apache.tika.metadata.Metadata();
+
+        ContentHandler tikaHandler = new ToTextContentHandler();
+        try {
+          parser.parse(is, tikaHandler, tikaMetadata, context);
+        } catch (Exception ex) {
+          throw new IOException(ex);
```

Comment:
Just make the processElement method throw Exception?

Reply:
Done